workers/unity/Assets/Gdk/Core/Collections/HandleProvider.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    internal static class HandleProvider
+    {
+        private static uint nextHandle = 0;
+        private static Dictionary<uint, Action<uint>> freeActions;
+
+        public static uint AllocateListHandle<TValue>()
```

Comment:
I would avoid the word allocate somewhere that doesn't do any allocation.

Reply:
In which case it should say that. Possibly call it AllocateList and return a handle.